File: CoffeeMaker/CSHARP/CoffeeMaker.Test/IntegrationTests/LetsIntegrationTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
namespace CoffeeMaker.IntegrationTests
{
    [TestClass]
    public class LetsIntegrationTests
    {
        public LetsIntegrationTests()
        {
        }
        [TestMethod]
        public void TestNoBeanCoffee()
        {
            // Arange
            Mock<ICoffeeSelect> CoffeeMock = new Mock<ICoffeeSelect>();
            CoffeeMock.Setup(test => test.GrindBeans()).Returns(true);
            CoffeeMock.Setup(test => test.GetBeans()).Returns(0);

            //make the coffee, notice no mock here.
            ICoffeeMaker coffeeMaker =
                                    new CoffeeMaker(CoffeeMock.Object);
            //Act
            string coffee = coffeeMaker.MakingCoffee();

            //Assert

            //verify 
            CoffeeMock.Verify(mock => mock.GrindBeans(),  Times.Never());
            CoffeeMock.Verify(mock => mock.GetBeans(), Times.Once());
            Assert.AreEqual(coffee, "No coffee available", "Failed because there are no beans");
        }

        [TestMethod]
        public void TestGrindFailure()
        {
            // Arange
            Mock<ICoffeeSelect> CoffeeMock = new Mock<ICoffeeSelect>();
            CoffeeMock.Setup(test => test.GrindBeans()).Returns(false);
            CoffeeMock.Setup(test => test.GetBeans()).Returns(1);

            //make the coffee, notice no mock here.
            ICoffeeMaker coffeeMaker =
                                    new CoffeeMaker(CoffeeMock.Object);
            //Act
            string coffee = coffeeMaker.MakingCoffee();

            //Assert
            CoffeeMock.Verify(mock => mock.GrindBeans(), Times.Once());
            CoffeeMock.Verify(mock => mock.GetBeans(), Times.Once());
            Assert.AreEqual(coffee, "No ground coffee available", "Grinder broke so no grinds");
        }

        [TestMethod]
        public void TestTheSweetSweetTasteOfCoffee()
        {
            // Arange
            Mock<ICoffeeSelect> CoffeeMock = new Mock<ICoffeeSelect>();
            CoffeeMock.Setup(test => test.GrindBeans()).Returns(true);
            CoffeeMock.Setup(test => test.GetBeans()).Returns(1);

            //make the coffee, notice no mock here.
            ICoffeeMaker coffeeMaker =
                                    new CoffeeMaker(CoffeeMock.Object);

            //Act
            string coffee = coffeeMaker.MakingCoffee();

            //Assert
            CoffeeMock.Verify(mock => mock.GrindBeans(), Times.Once());
            CoffeeMock.Verify(mock => mock.GetBeans(), Times.Once());
            Assert.AreEqual(coffee, "Coffee has now been brewed", "Yum we have coffee");
        }
    }
}

File: Mocking/CoffeeMaker/Java/src/test/java/LetsIntegrationTests.java
import Interfaces.*;
import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;
import org.junit.jupiter.api.Test;
import jdk.jshell.spi.ExecutionControl;


public class LetsIntegrationTests {

    @Test
    public void TestNoBeanCoffee() throws ExecutionControl.NotImplementedException {
        ICoffeeSelect coffeeMakerMock = mock(ICoffeeSelect.class);
        when(coffeeMakerMock.GetBeans()).thenReturn(0);
        when(coffeeMakerMock.GrindBeans()).thenReturn(true);

        ICoffeeMaker coffeemaker = new CoffeeMaker(coffeeMakerMock);
        String coffee = coffeemaker.MakingCoffee();
        assertEquals(coffee, "No coffee available");
        verify(coffeeMakerMock, times(1)).GetBeans();
        verify(coffeeMakerMock, times(0)).GrindBeans();
    }

    @Test
    public void TestGrindFailure() throws ExecutionControl.NotImplementedException {
        ICoffeeSelect coffeeMakerMock = mock(ICoffeeSelect.class);
        when(coffeeMakerMock.GetBeans()).thenReturn(1);
        when(coffeeMakerMock.GrindBeans()).thenReturn(false);

        ICoffeeMaker coffeemaker = new CoffeeMaker(coffeeMakerMock);
        String coffee = coffeemaker.MakingCoffee();
        assertEquals(coffee, "No ground coffee available");
        verify(coffeeMakerMock, times(1)).GetBeans();
        verify(coffeeMakerMock, times(1)).GrindBeans();
    }

    @Test
    public void TestTheSweetSweetTasteOfCoffee() throws ExecutionControl.NotImplementedException {
         ICoffeeSelect coffeeMakerMock = mock(ICoffeeSelect.class);
         when(coffeeMakerMock.GetBeans()).thenReturn(1);
         when(coffeeMakerMock.GrindBeans()).thenReturn(true);

        ICoffeeMaker coffeemaker = new CoffeeMaker(coffeeMakerMock);
        String coffee = coffeemaker.MakingCoffee();
        assertEquals(coffee, "Coffee has now been brewed");
        verify(coffeeMakerMock, times(1)).GetBeans();
        verify(coffeeMakerMock, times(1)).GrindBeans();
    }
}
